pandas/io/tests/test_parsers.py

```diff
@@ -2679,6 +2679,27 @@ def test_fwf_compression(self):
                                   compression=comp_name)
                 tm.assert_frame_equal(result, expected)
 
+    def test_fwf_for_uint8(self):
+        data = """1421302965.213420    PRI=3 PGN=0xef00      DST=0x17 SRC=0x28    04 154 00 00 00 00 00 127
+1421302964.226776    PRI=6 PGN=0xf002               SRC=0x47    243 00 00 255 247 00 00 71"""
+        df = read_fwf(StringIO(data), 
+                colspecs=[(0,17),(25,26),(33,37),(49,51),(58,62),(63,1000)],
+                names=['time','pri','pgn','dst','src','data'],
+                converters={
+                        'pgn':lambda x: int(x,16),
+                        'src':lambda x: int(x,16),
+                        'dst':lambda x: int(x,16),
+                        'data':lambda x: len(x.split(' '))})
+
+        expected = DataFrame([[1421302965.213420,3,61184,23,40,8], 
+                [1421302964.226776,6,61442,None, 71,8]],
```

Comment:
@jreback — so still halfway trying to figure this out but adding the `dtype` argument to `read_fwf` returns an engine not support error even when I set the engine to the C engine. 

I'm not entirely certain what effect adding the `dtype` will have on the tests as it appears that the error is thrown when parsing the dictionary in the `converters` parameter. Specifically for this case, the `except` clause is trigger when parsing the values associated with `DST` since only one of them is provided in the test data and the conversion from hex to decimal on [line 996](https://github.com/pydata/pandas/blob/master/pandas/io/parsers.py#L996) fails.

Let me know if I am not understanding something.